org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodAnalyzer.java

```diff
@@ -86,6 +99,38 @@ public IMethodCoverage getCoverage() {
 		return coverage;
 	}
 
+	/**
+	 * {@link MethodNode#accept(MethodVisitor)}
+	 */
+	@Override
+	public void accept(final MethodNode methodNode,
+			final MethodVisitor methodVisitor) {
+		this.ignored.clear();
+		for (final IFilter filter : FILTERS) {
+			filter.filter(methodNode, this);
+		}
+
+		for (final TryCatchBlockNode n : methodNode.tryCatchBlocks) {
+			n.accept(methodVisitor);
+		}
+		for (int i = 0; i < methodNode.instructions.size(); i++) {
```

Comment:
Only suggestions that come to mind are:
1. Add build-time class file scanning for use of `InsnList.get(int)`.
2. Create `InsnList` subclass that throws and use it in unittests.  I guess this would require subclassing `ClassAnalyzer` to override `visitMethod`, which probably requires reimplementing `Analyzer.analyzeClass`/`createAnalyzingVisitor`.